### PR TITLE
Fix: Remove deleted tags from items

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- None

**Details**
- When a tag is deleted, it is removed from all files
- When a tag is deleted/updated, the current page is refreshed to show changes
- Added a null check to avoid `NullReference` when the user hits `Esc` after deleting a tag

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Comment by yaira2** [comment]
It's removing it from the alternate data stream?

**Comment by ferrariofilippo** [comment]
> It's removing it from the alternate data stream?

I can't test this, I can't tag ADS. Whenever I apply a tag, refreshing the page will remove it

**Comment by yaira2** [comment]
I think we should wait for @gave92 to review this one.

File: src/Files.App/BaseLayout.cs [hunk 1]
```diff
@@ -308,15 +308,22 @@ public BaseLayout()
 		private void HookBaseEvents()
 		{
 			ItemManipulationModel.RefreshItemsOpacityInvoked += ItemManipulationModel_RefreshItemsOpacityInvoked;
+			FileTagsSettingsService.OnTagsUpdated += FileTagsSettingsService_OnTagsUpdated;
```
Comment (gave92) [inline src/Files.App/BaseLayout.cs:311]: Refresh is usually done from FilesystemViewModel.

File: src/Files.App/ServicesImplementation/Settings/FileTagsSettingsService.cs [hunk 2]
```diff
@@ -107,13 +108,17 @@ public void EditTag(string uid, string name, string color)
...
+			var untagFilesTask = UntagAllFiles(uid);
+
 			var (_, index) = GetTagAndIndex(uid);
 			if (index == -1)
 				return;
 
 			var oldTags = FileTagList.ToList();
 			oldTags.RemoveAt(index);
 			FileTagList = oldTags;
+			untagFilesTask.Wait();
```
Comment (gave92) [inline src/Files.App/ServicesImplementation/Settings/FileTagsSettingsService.cs:120]: Can't we make this method async instead?

File: src/Files.App/ServicesImplementation/Settings/FileTagsSettingsService.cs [hunk 3]
```diff
@@ -163,5 +168,25 @@ public override object ExportSettings()
...
 		}
+
+		private Task UntagAllFiles(string uid)
+		{
+			foreach (var item in FileTagsHelper.GetDbInstance().GetAll())
+			{
+				if (item.Tags.Contains(uid))
+				{
+					var newTags = new string[item.Tags.Length - 1];
+					for (int i = 0, ii = 0; i < item.Tags.Length; i++)
```
Comment (gave92) [inline src/Files.App/ServicesImplementation/Settings/FileTagsSettingsService.cs:179]: This can probably be made more readable by using Linq.

**Comment by gave92** [comment]
> It's removing it from the alternate data stream?

Yeah it should

**Comment by gave92** [comment]
Can't review anytime soon but looks good

**Comment by yaira2** [comment]
> Can't review anytime soon but looks good

A looks good from you is enough for me 😂

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by ferrariofilippo** [comment]
> LGTM

That was speedy!!!!

**Pull request merged**